cogs/obliterate.py

```diff
@@ -254,6 +255,68 @@ async def on_error(self, interaction: discord.Interaction, error: Exception):
         print(error)
         traceback.print_tb(error.__traceback__)
 
+class AppreciationModal(discord.ui.Modal, title="Message of Appreciation"):
+    def __init__(self, bot, appreciate_players):
+        super().__init__()
+        self.bot = bot
+        self.appreciate_player = appreciate_players
+
+    message = discord.ui.TextInput(label='Message of appreciation', min_length=1, max_length=200, required=True, style=TextStyle.short)
+    anonymous = discord.ui.TextInput(label='Would you like this post to be anonymous?', min_length=1, max_length=3, required=True, placeholder="Yes/No", style=TextStyle.short)
+
+    async def on_submit(self, interaction: discord.Interaction):
+        user_id = int(interaction.message.embeds[0].footer.text.replace('User ID: ', ''))
+        member = await interaction.guild.fetch_member(user_id)
+        message = self.message.value
+        anonymous = self.anonymous.value
+        appreciate_players = self.appreciate_players
+
+        if not anonymous.upper() in ['NO', 'N', 'YES', 'Y']:
+            return f'Error invalid value for anonymous: `{anonymous}`'
+        else:
+            for i, opt in enumerate(['NO', 'N', 'YES', 'Y']):
+                if anonymous.upper() == opt and i < 2:
+                    anonymous = False
+                    break
+                elif anonymous.upper() == opt:
+                    anonymous = True
+                    break
+
+        if not member:
+            return 'Error: user not found'
+
+        # Update appreciation sheet on roster
+        agc = await self.bot.agcm.authorize()
+        ss = await agc.open_by_key(config['obliterate_roster_key'])
+        roster = await ss.worksheet('Appreciation')
+
+        members_col = await roster.col_values(1)
+        rows = len(members_col)
+
+        date_str = datetime.utcnow().strftime('%d %b %Y')
+        date_str = date_str if not date_str.startswith('0') else date_str[1:]
+        new_row = [member, appreciate_players, message, date_str]
+        cell_list = [gspread.models.Cell(rows+1, i+1, value=val) for i, val in enumerate(new_row)]
+        print(f'writing values:\n{new_row}\nto row {rows+1}')
+        await roster.update_cells(cell_list, nowait=True)
+
+        # Send appreciation message to Appreciation Station channel
+        players = ' '.join(str(n) for n in appreciate_players)
+        appreciation_message = f'Appreciation for: {players}\n'
+        appreciation_message += f'{message}'
```

Comment:
I would use an embed here for a nice-looking result

Reply:
Would the embed be able to tag the `appreciate_player` still? Or would we keep the player tag outside of the embed, and keep the embed just for the message and user who sent it?

Reply:
Members can be mentioned inside an embed, however, they will not be notified. But if notifications are required, it is possible to send both an embed and a regular text message (a mention, in this case) in one single message. You would do this as follows:
```py
await interaction.response.send_message(member.mention, embed=embed)
```

Reply:
If we decide not to notify the user, then we could probably just add a text input to enter the user's name manually instead of the autocomplete. I'm thinking of a situation where someone is helped by someone while on their alt, and the player giving the appreciation may not be able to tag them as they don't know their disc name which could be set as the main account's name? (I hope that made sense! 😂 )
Also could start getting annoying if people are always getting tagged from these messages. Should we maybe just ditch the tagging idea and allow the user to enter the rsn of the person who they are appreciating?

Reply:
Even if we don't intend to notify them, it is still important to make sure we get a correct user. I believe the intention was to use these appreciations as a metric to earn points towards rank promotions, so we have to be able to link these to some clan member regardless. We also wouldn't want points going to an alt instead of someone's main